raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -317,31 +312,66 @@ def test_unsubscribe_from_invalid(comms_nodes):
     assert "not subscribed to" in str.lower(e.value.details())
 
 
-@pytest.mark.parametrize("amount_of_nodes", [1])
-def test_unsubscribe_from_self(comms_nodes):
-    client, address = comms_nodes[1].client, comms_nodes[1].address
+@pytest.mark.parametrize("nodes_to_clients", [{1: 1, 2: 1}, {1: 2}])
+@pytest.mark.xfail(reason="wrong rif comms subscription behaviour cluster 2")
+def test_unsubscribe_from_self(comms_clients):
+    client_1 = comms_clients[1]
+    client_2 = comms_clients[2]
 
     # subscribe to self, then unsubscribe from self
-    client.subscribe_to(address)
-    client.unsubscribe_from(address)
+    client_1.subscribe_to(client_1.rsk_address.address)
+    assert client_1._is_subscribed_to(client_1.rsk_address.address) is True
+    assert client_1._is_subscribed_to(client_2.rsk_address.address) is False
+
+    client_2.subscribe_to(client_2.rsk_address.address)
+    assert client_2._is_subscribed_to(client_2.rsk_address.address) is True
+    assert client_2._is_subscribed_to(client_1.rsk_address.address) is False  # FIXME this fails scenario cluster 2
```

Comment:
since this is a `self`-kind of test, clients should be iterated through using a `for` sentence

Reply:
apologies, my original comment was meant for an older version of the test.

thanks for your comment, you are correct :+1: